src/argilla/client/client.py

```diff
@@ -365,15 +365,23 @@ def log(
         if record_type is TextClassificationRecord:
             bulk_class = TextClassificationBulkData
             creation_class = CreationTextClassificationRecord
+            task = TaskType.text_classification
         elif record_type is TokenClassificationRecord:
             bulk_class = TokenClassificationBulkData
             creation_class = CreationTokenClassificationRecord
+            task = TaskType.token_classification
         elif record_type is Text2TextRecord:
             bulk_class = Text2TextBulkData
             creation_class = CreationText2TextRecord
+            task = TaskType.text2text
         else:
             raise InputValueError(f"Unknown record type {record_type}. Available values are {Record.__args__}")
 
+        try:
+            self.datasets.create(name=name, task=task, workspace=workspace)
```

Comment:
We should include the workspace back to be able to create the dataset in a workspace, as if there's none, then the default will be set